src/Totem/ClientApp/features/contracts/AddModelModalWindow.vue

```diff
@@ -166,7 +166,19 @@ export default {
 
     deleteModel() {
       const model = deepCopy(last(this.editStack));
-      this.$emit('delete', model);
+      if (this.editStack.length - 1 > 0) {
+        const parent = this.editStack[this.editStack.length - 2];
```

Comment:
This doesn't work if the model deleted is not an immediate child of the parent window. See example where 2x nested model is removed from the root parent:
![bug-22](https://user-images.githubusercontent.com/3969554/65080705-86ec4200-d967-11e9-9fa2-1496e6cc49a9.gif)